driver/reduceextreme_driver.hpp

```diff
@@ -0,0 +1,510 @@
+
+        int32_t extreme_idx = 0;
+        Tcheck extreme      = static_cast<Tcheck>(x[x_idx]);
+
+        for(int32_t i = 1; i < reduce_size; ++i)
+        {
+            x_idx += inner_size;
+            Tcheck val = static_cast<Tcheck>(x[x_idx]);
+            reduce_func<Tcheck, int32_t, op>{}.calculate(extreme, val, extreme_idx, i);
+        }
+        indicehost[o] = extreme_idx;
+        if(yhost)
+            yhost[o] = extreme;
+    }
+    return ret;
+}
+
+template <typename Tgpu, typename Tref>
+class ReduceExtremeDriver : public Driver
+{
+public:
+    ReduceExtremeDriver() : Driver()
+    {
+        miopenCreateTensorDescriptor(&xDesc);
+        miopenCreateTensorDescriptor(&yDesc);
+        miopenCreateTensorDescriptor(&indiceDesc);
+
+        data_type        = miopen_type<Tgpu>{};
+        indice_data_type = miopen_type<int32_t>{};
+    }
+
+    int AddCmdLineArgs() override;
+    int ParseCmdLineArgs(int argc, char* argv[]) override;
+    InputFlags& GetInputFlags() override { return inflags; }
+
+    int GetandSetData() override;
+
+    int AllocateBuffersAndCopy() override;
+
+    int RunForwardGPU() override;
+    int RunForwardCPU();
+
+    int RunBackwardGPU() override;
+
+    Tref GetTolerance();
+    int VerifyBackward() override;
+    int VerifyForward() override;
+    ~ReduceExtremeDriver() override
+    {
+        miopenDestroyTensorDescriptor(xDesc);
+        miopenDestroyTensorDescriptor(yDesc);
+        miopenDestroyTensorDescriptor(indiceDesc);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+
+    miopenTensorDescriptor_t xDesc;
+    miopenTensorDescriptor_t yDesc;
+    miopenTensorDescriptor_t indiceDesc;
+
+    std::unique_ptr<GPUMem> x_dev;
+    std::unique_ptr<GPUMem> indice_dev;
+    std::unique_ptr<GPUMem> y_dev;
+
+    std::vector<Tgpu> x;
+    std::vector<Tgpu> y;
+    std::vector<Tref> yhost;
+    std::vector<int32_t> indice;
+    std::vector<int32_t> indicehost;
+
+    int dim;
+    miopenReduceExtremeOp_t reduceExtremeOp;
+
+    miopenDataType_t indice_data_type;
+};
+
+template <typename Tgpu, typename Tref>
+int ReduceExtremeDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+
+    if((static_cast<miopenReduceExtremeOp_t>(inflags.GetValueInt("ReduceExtremeOp")) < 1) ||
+       (static_cast<miopenReduceExtremeOp_t>(inflags.GetValueInt("ReduceExtremeOp")) > 4))
+        std::cerr << "Error ReduceExtremeOp(1-4)" << std::endl;
+
+    auto inTensorParam = inflags.GetValueTensor("input");
+
+    if((static_cast<miopenReduceExtremeOp_t>(inflags.GetValueInt("DimToReduce")) < 0) ||
+       (static_cast<miopenReduceExtremeOp_t>(inflags.GetValueInt("DimToReduce")) >
+        inTensorParam.lengths.size() - 1))
+        std::cerr << "Error DimToReduce(0-" << inTensorParam.lengths.size() - 1 << ")" << std::endl;
+
+    return miopenStatusSuccess;
```

Comment:
leftover:
> return a non-zero value if DimToReduce is wrong.

Also please return error code if ReduceExtremeOp is wrong

Reply:
fixed it.